 # Project: Bank Churn Data Exploration and Prediction

## Table of Contents
 <ul>
 <li><a href="#intro">Introduction</a></li>
 <li><a href="#wrangling">Data Wrangling</a></li>
 <li><a href="#eda">Exploratory Data Analysis</a></li>
 <li><a href="#Machine Learnning model">Machine Learnning model</a></li>
 </ul>

# <a id='intro'></a>
## Introduction
**In the competitive world of banking, keeping customers happy is crucial. The "Bank Churn Prediction" project is all about using smart technology to understand customer behavior and predict if someone might leave the bank. By looking at past data and using advanced tools, we aim to help the bank take proactive steps to keep customers from leaving.  ultimately leading to happier customers and a more successful bank.This data set contains information about Bank Churn it includes 10127 rows and 23 columns.**

*Before the analysis of the dataset, data wrangling phase has been conducted to clean the data from unimportant columns, noisy data, and other problems. Before data wrangling phase, general properities about the dataset has been addressed.*
 
## Analysis of the DataSet (questions that is analyized in the dataset):
1- What percentage of customers in the dataset have experienced attrition

2- How does the distribution of 'Avg_Utilization_Ratio' differ between attrited and existing customers based on the box plot

3- Are there any income groups that tend to have higher or lower total revolving balances

4- How does the number of dependents vary across different marital statuses

5- Is there a correlation between the type of card a customer holds and their income category

6- What is the relationship between Credit_Limit  and Avg_Open_To_Buy

7- What insights can we derive about customer behavior from the box plot of total transactions?

8- How is the correlation between the number of total transactions ('Total_Trans_Ct') and the total transaction amount ('Total_Trans_Amt')?

9- How does the distribution of 'Credit_Limit' vary across different levels of 'Avg_Open_To_Buy?

10-How does the number of inactive months ('Months_Inactive_12_mon') vary over time?

11-What is the Positive Correlation And the Negative Correlation between numerical column?

12-What is the relationship between Total_Trans_Amt and Total_Trans_Ct

13-How is the distribution of a specific variable count related to the age of customers?

14- What is the relationship between Total_Trans_Amt  and Total_Trans_Ct

15- What is the distribution of card categories among customers with different education levels?

16- How does the distribution of customer age vary across different income categories?

In [3]:
# import packages to use
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sn
import plotly.express as px
import plotly.graph_objects as go

## <a id='wrangling'></a>
## Data Wrangling
 *Tip*: In this section of the report, you will load in the data, check for cleanliness, and then trim and clean your dataset for analysis. Make sure that you document your steps carefully and justify your cleaning decisions. We will start with addressing General properities about the dataset.

### General Properties

In [7]:
import pandas as pd
# Load csv file named BankChurners.csv into about our bank set
bank = pd.read_csv(r"D:\BankChurners.csv")

# dimensions of our bank set
print("Dimensions of bank: ", bank.shape)

# print Information of our bank set
print("\nInformation about bank: ")
print(bank.info())

# First ten rows of our bank data set
print("\nHead of bank: \n",(bank.head(5)))

Dimensions of bank:  (10127, 23)

Information about bank: 
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10127 entries, 0 to 10126
Data columns (total 23 columns):
 #   Column                                                                                                                              Non-Null Count  Dtype  
---  ------                                                                                                                              --------------  -----  
 0   CLIENTNUM                                                                                                                           10127 non-null  int64  
 1   Attrition_Flag                                                                                                                      10127 non-null  object 
 2   Customer_Age                                                                                                                        10127 non-null  int64  
 3   Gender                

## Data Cleaning
Data cleaning is the process of fixing or removing incorrect, duplicate,noisy, or incomplete data within a dataset. 
Thorugh data cleaning phase, data wrangling has been conducted for modifying the data into a more usable form.

#### Problems With the dataset
>1: Remove unused Columns ['Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1', 'Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_2']
>
>2: Remove duplication in the rows (check and process if  exist)
>
>3: Unknown data have been replaced by the most frequent data used "Graduated" in education level 
>
>4: In marital status the unknown input has been replaced by "Married" which is the most frequent data used
>
>5: Modify the datatype of 'Credit_Limit' column to the correct datatype (from float into int64)


In [8]:
bank.isnull().sum()

CLIENTNUM                                                                                                                             0
Attrition_Flag                                                                                                                        0
Customer_Age                                                                                                                          0
Gender                                                                                                                                0
Dependent_count                                                                                                                       0
Education_Level                                                                                                                       0
Marital_Status                                                                                                                        0
Income_Category                                 

#### 1: Remove unused Columns 

In [9]:
# drop Columns ()
bank.drop([ 'Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_1', 'Naive_Bayes_Classifier_Attrition_Flag_Card_Category_Contacts_Count_12_mon_Dependent_count_Education_Level_Months_Inactive_12_mon_2'], axis=1, inplace=True)

#print Columns after removing the two columns
print(bank.columns.tolist())


['CLIENTNUM', 'Attrition_Flag', 'Customer_Age', 'Gender', 'Dependent_count', 'Education_Level', 'Marital_Status', 'Income_Category', 'Card_Category', 'Months_on_book', 'Total_Relationship_Count', 'Months_Inactive_12_mon', 'Contacts_Count_12_mon', 'Credit_Limit', 'Total_Revolving_Bal', 'Avg_Open_To_Buy', 'Total_Amt_Chng_Q4_Q1', 'Total_Trans_Amt', 'Total_Trans_Ct', 'Total_Ct_Chng_Q4_Q1', 'Avg_Utilization_Ratio']


#### 2: Remove duplication in the rows (check and process if exist)

In [10]:
# Number of rows 
print("Current Number of rows", bank.shape[0])
# check Rows Duplication
duplicated_rows = sum(bank.duplicated())
print("Number of Duplicated rows: ", duplicated_rows)
if(duplicated_rows):
    bank.drop_duplicates(keep ='first', inplace=True)
print("New Number of rows after Removing duplication",(bank.shape[0]))



Current Number of rows 10127
Number of Duplicated rows:  0
New Number of rows after Removing duplication 10127


In [11]:
df = pd.DataFrame(bank)
# Display the original DataFrame
print("Original DataFrame:")
print(df)
# Replace missing values with the most frequent value in each column
df_filled = df.apply(lambda col: col.fillna(col.mode()[0]))
# Display the DataFrame after filling missing values
print("\nDataFrame after filling missing values:")
print(df_filled)


Original DataFrame:
       CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
0      768805383  Existing Customer            45      M                3   
1      818770008  Existing Customer            49      F                5   
2      713982108  Existing Customer            51      M                3   
3      769911858  Existing Customer            40      F                4   
4      709106358  Existing Customer            40      M                3   
...          ...                ...           ...    ...              ...   
10122  772366833  Existing Customer            50      M                2   
10123  710638233  Attrited Customer            41      M                2   
10124  716506083  Attrited Customer            44      F                1   
10125  717406983  Attrited Customer            30      M                2   
10126  714337233  Attrited Customer            43      F                2   

      Education_Level Marital_Status Income_Category Ca

In [12]:
df = pd.DataFrame(bank)
# Find the most frequent value in each column
most_frequent_values = df.mode().iloc[0]

print("Most frequent values in each column:")
print(most_frequent_values)


Most frequent values in each column:
CLIENTNUM                           708082083
Attrition_Flag              Existing Customer
Customer_Age                             44.0
Gender                                      F
Dependent_count                           3.0
Education_Level                      Graduate
Marital_Status                        Married
Income_Category                Less than $40K
Card_Category                            Blue
Months_on_book                           36.0
Total_Relationship_Count                  3.0
Months_Inactive_12_mon                    3.0
Contacts_Count_12_mon                     3.0
Credit_Limit                          34516.0
Total_Revolving_Bal                       0.0
Avg_Open_To_Buy                        1438.3
Total_Amt_Chng_Q4_Q1                    0.791
Total_Trans_Amt                        4253.0
Total_Trans_Ct                           81.0
Total_Ct_Chng_Q4_Q1                     0.667
Avg_Utilization_Ratio                     0

#### 3: Unknown data have been replaced by the most frequent data used "Graduated" in education level 

In [13]:
bank["Education_Level"]=bank["Education_Level"].str.replace('Unknown','Graduate')
bank

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
0      768805383  Existing Customer            45      M                3   
1      818770008  Existing Customer            49      F                5   
2      713982108  Existing Customer            51      M                3   
3      769911858  Existing Customer            40      F                4   
4      709106358  Existing Customer            40      M                3   
...          ...                ...           ...    ...              ...   
10122  772366833  Existing Customer            50      M                2   
10123  710638233  Attrited Customer            41      M                2   
10124  716506083  Attrited Customer            44      F                1   
10125  717406983  Attrited Customer            30      M                2   
10126  714337233  Attrited Customer            43      F                2   

      Education_Level Marital_Status Income_Category Card_Category  \
0         High School        Married     $60K - $80K          Blue   
1            Graduate         Single  Less than $40K          Blue   
2            Graduate        Married    $80K - $120K          Blue   
3         High School        Unknown  Less than $40K          Blue   
4          Uneducated        Married     $60K - $80K          Blue   
...               ...            ...             ...           ...   
10122        Graduate         Single     $40K - $60K          Blue   
10123        Graduate       Divorced     $40K - $60K          Blue   
10124     High School        Married  Less than $40K          Blue   
10125        Graduate        Unknown     $40K - $60K          Blue   
10126        Graduate        Married  Less than $40K        Silver   

       Months_on_book  ...  Months_Inactive_12_mon  Contacts_Count_12_mon  \
0                  39  ...                       1                      3   
1                  44  ...                       1                      2   
2                  36  ...                       1                      0   
3                  34  ...                       4                      1   
4                  21  ...                       1                      0   
...               ...  ...                     ...                    ...   
10122              40  ...                       2                      3   
10123              25  ...                       2                      3   
10124              36  ...                       3                      4   
10125              36  ...                       3                      3   
10126              25  ...                       2                      4   

       Credit_Limit  Total_Revolving_Bal  Avg_Open_To_Buy  \
0           12691.0                  777          11914.0   
1            8256.0                  864           7392.0   
2            3418.0                    0           3418.0   
3            3313.0                 2517            796.0   
4            4716.0                    0           4716.0   
...             ...                  ...              ...   
10122        4003.0                 1851           2152.0   
10123        4277.0                 2186           2091.0   
10124        5409.0                    0           5409.0   
10125        5281.0                    0           5281.0   
10126       10388.0                 1961           8427.0   

       Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
0                     1.335             1144              42   
1                     1.541             1291              33   
2                     2.594             1887              20   
3                     1.405             1171              20   
4                     2.175              816              28   
...                     ...              ...             ...   
10122                 0.703            15476             117   
10123                 0.804             8764              69   
10124  

#### 4: In marital status the unknown input has been replaced by "Married" which is the most frequent data used

In [14]:
bank["Income_Category"]=bank["Income_Category"].str.replace('Unknown','Less than $40K')
bank

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
0      768805383  Existing Customer            45      M                3   
1      818770008  Existing Customer            49      F                5   
2      713982108  Existing Customer            51      M                3   
3      769911858  Existing Customer            40      F                4   
4      709106358  Existing Customer            40      M                3   
...          ...                ...           ...    ...              ...   
10122  772366833  Existing Customer            50      M                2   
10123  710638233  Attrited Customer            41      M                2   
10124  716506083  Attrited Customer            44      F                1   
10125  717406983  Attrited Customer            30      M                2   
10126  714337233  Attrited Customer            43      F                2   

      Education_Level Marital_Status Income_Category Card_Category  \
0         High School        Married     $60K - $80K          Blue   
1            Graduate         Single  Less than $40K          Blue   
2            Graduate        Married    $80K - $120K          Blue   
3         High School        Unknown  Less than $40K          Blue   
4          Uneducated        Married     $60K - $80K          Blue   
...               ...            ...             ...           ...   
10122        Graduate         Single     $40K - $60K          Blue   
10123        Graduate       Divorced     $40K - $60K          Blue   
10124     High School        Married  Less than $40K          Blue   
10125        Graduate        Unknown     $40K - $60K          Blue   
10126        Graduate        Married  Less than $40K        Silver   

       Months_on_book  ...  Months_Inactive_12_mon  Contacts_Count_12_mon  \
0                  39  ...                       1                      3   
1                  44  ...                       1                      2   
2                  36  ...                       1                      0   
3                  34  ...                       4                      1   
4                  21  ...                       1                      0   
...               ...  ...                     ...                    ...   
10122              40  ...                       2                      3   
10123              25  ...                       2                      3   
10124              36  ...                       3                      4   
10125              36  ...                       3                      3   
10126              25  ...                       2                      4   

       Credit_Limit  Total_Revolving_Bal  Avg_Open_To_Buy  \
0           12691.0                  777          11914.0   
1            8256.0                  864           7392.0   
2            3418.0                    0           3418.0   
3            3313.0                 2517            796.0   
4            4716.0                    0           4716.0   
...             ...                  ...              ...   
10122        4003.0                 1851           2152.0   
10123        4277.0                 2186           2091.0   
10124        5409.0                    0           5409.0   
10125        5281.0                    0           5281.0   
10126       10388.0                 1961           8427.0   

       Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
0                     1.335             1144              42   
1                     1.541             1291              33   
2                     2.594             1887              20   
3                     1.405             1171              20   
4                     2.175              816              28   
...                     ...              ...             ...   
10122                 0.703            15476             117   
10123                 0.804             8764              69   
10124  

In [15]:
bank["Marital_Status"]=bank["Marital_Status"].str.replace('Unknown','Married')
bank

CLIENTNUM     Attrition_Flag  Customer_Age Gender  Dependent_count  \
0      768805383  Existing Customer            45      M                3   
1      818770008  Existing Customer            49      F                5   
2      713982108  Existing Customer            51      M                3   
3      769911858  Existing Customer            40      F                4   
4      709106358  Existing Customer            40      M                3   
...          ...                ...           ...    ...              ...   
10122  772366833  Existing Customer            50      M                2   
10123  710638233  Attrited Customer            41      M                2   
10124  716506083  Attrited Customer            44      F                1   
10125  717406983  Attrited Customer            30      M                2   
10126  714337233  Attrited Customer            43      F                2   

      Education_Level Marital_Status Income_Category Card_Category  \
0         High School        Married     $60K - $80K          Blue   
1            Graduate         Single  Less than $40K          Blue   
2            Graduate        Married    $80K - $120K          Blue   
3         High School        Married  Less than $40K          Blue   
4          Uneducated        Married     $60K - $80K          Blue   
...               ...            ...             ...           ...   
10122        Graduate         Single     $40K - $60K          Blue   
10123        Graduate       Divorced     $40K - $60K          Blue   
10124     High School        Married  Less than $40K          Blue   
10125        Graduate        Married     $40K - $60K          Blue   
10126        Graduate        Married  Less than $40K        Silver   

       Months_on_book  ...  Months_Inactive_12_mon  Contacts_Count_12_mon  \
0                  39  ...                       1                      3   
1                  44  ...                       1                      2   
2                  36  ...                       1                      0   
3                  34  ...                       4                      1   
4                  21  ...                       1                      0   
...               ...  ...                     ...                    ...   
10122              40  ...                       2                      3   
10123              25  ...                       2                      3   
10124              36  ...                       3                      4   
10125              36  ...                       3                      3   
10126              25  ...                       2                      4   

       Credit_Limit  Total_Revolving_Bal  Avg_Open_To_Buy  \
0           12691.0                  777          11914.0   
1            8256.0                  864           7392.0   
2            3418.0                    0           3418.0   
3            3313.0                 2517            796.0   
4            4716.0                    0           4716.0   
...             ...                  ...              ...   
10122        4003.0                 1851           2152.0   
10123        4277.0                 2186           2091.0   
10124        5409.0                    0           5409.0   
10125        5281.0                    0           5281.0   
10126       10388.0                 1961           8427.0   

       Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
0                     1.335             1144              42   
1                     1.541             1291              33   
2                     2.594             1887              20   
3                     1.405             1171              20   
4                     2.175              816              28   
...                     ...              ...             ...   
10122                 0.703            15476             117   
10123                 0.804             8764              69   
10124  

In [16]:
bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10127 entries, 0 to 10126
Data columns (total 21 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   CLIENTNUM                 10127 non-null  int64  
 1   Attrition_Flag            10127 non-null  object 
 2   Customer_Age              10127 non-null  int64  
 3   Gender                    10127 non-null  object 
 4   Dependent_count           10127 non-null  int64  
 5   Education_Level           10127 non-null  object 
 6   Marital_Status            10127 non-null  object 
 7   Income_Category           10127 non-null  object 
 8   Card_Category             10127 non-null  object 
 9   Months_on_book            10127 non-null  int64  
 10  Total_Relationship_Count  10127 non-null  int64  
 11  Months_Inactive_12_mon    10127 non-null  int64  
 12  Contacts_Count_12_mon     10127 non-null  int64  
 13  Credit_Limit              10127 non-null  float64
 14  Total_

#### 5: Modify the datatype of 'Credit_Limit' column to the correct datatype (from float into int64)

In [17]:
# convert the 'Credit_Limit' column to int format
bank['Credit_Limit'] = bank['Credit_Limit'].astype('int64')

# Check the format of ' Credit_Limit ' column
bank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10127 entries, 0 to 10126
Data columns (total 21 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   CLIENTNUM                 10127 non-null  int64  
 1   Attrition_Flag            10127 non-null  object 
 2   Customer_Age              10127 non-null  int64  
 3   Gender                    10127 non-null  object 
 4   Dependent_count           10127 non-null  int64  
 5   Education_Level           10127 non-null  object 
 6   Marital_Status            10127 non-null  object 
 7   Income_Category           10127 non-null  object 
 8   Card_Category             10127 non-null  object 
 9   Months_on_book            10127 non-null  int64  
 10  Total_Relationship_Count  10127 non-null  int64  
 11  Months_Inactive_12_mon    10127 non-null  int64  
 12  Contacts_Count_12_mon     10127 non-null  int64  
 13  Credit_Limit              10127 non-null  int64  
 14  Total_

In [18]:
bank.to_csv('bank.csv', sep=",")

In [20]:
# Assuming 'bank.csv' is your CSV file
csv_file_path = r'D:\Bank.csv'

# Read the CSV file into a DataFrame
Bank = pd.read_csv(csv_file_path)

# Now, 'Bank' is a DataFrame containing the data from the CSV file

In [21]:
Bank

Unnamed: 0  CLIENTNUM     Attrition_Flag  Customer_Age Gender  \
0               0  768805383  Existing Customer            45      M   
1               1  818770008  Existing Customer            49      F   
2               2  713982108  Existing Customer            51      M   
3               3  769911858  Existing Customer            40      F   
4               4  709106358  Existing Customer            40      M   
...           ...        ...                ...           ...    ...   
10122       10122  772366833  Existing Customer            50      M   
10123       10123  710638233  Attrited Customer            41      M   
10124       10124  716506083  Attrited Customer            44      F   
10125       10125  717406983  Attrited Customer            30      M   
10126       10126  714337233  Attrited Customer            43      F   

       Dependent_count Education_Level Marital_Status Income_Category  \
0                    3     High School        Married     $60K - $80K   
1                    5        Graduate         Single  Less than $40K   
2                    3        Graduate        Married    $80K - $120K   
3                    4     High School        Married  Less than $40K   
4                    3      Uneducated        Married     $60K - $80K   
...                ...             ...            ...             ...   
10122                2        Graduate         Single     $40K - $60K   
10123                2        Graduate       Divorced     $40K - $60K   
10124                1     High School        Married  Less than $40K   
10125                2        Graduate        Married     $40K - $60K   
10126                2        Graduate        Married  Less than $40K   

      Card_Category  ...  Months_Inactive_12_mon  Contacts_Count_12_mon  \
0              Blue  ...                       1                      3   
1              Blue  ...                       1                      2   
2              Blue  ...                       1                      0   
3              Blue  ...                       4                      1   
4              Blue  ...                       1                      0   
...             ...  ...                     ...                    ...   
10122          Blue  ...                       2                      3   
10123          Blue  ...                       2                      3   
10124          Blue  ...                       3                      4   
10125          Blue  ...                       3                      3   
10126        Silver  ...                       2                      4   

       Credit_Limit  Total_Revolving_Bal  Avg_Open_To_Buy  \
0             12691                  777          11914.0   
1              8256                  864           7392.0   
2              3418                    0           3418.0   
3              3313                 2517            796.0   
4              4716                    0           4716.0   
...             ...                  ...              ...   
10122          4003                 1851           2152.0   
10123          4277                 2186           2091.0   
10124          5409                    0           5409.0   
10125          5281                    0           5281.0   
10126         10388                 1961           8427.0   

       Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
0                     1.335             1144              42   
1                     1.541             1291              33   
2                     2.594             1887              20   
3                     1.405             1171              20   
4                     2.175              816              28   
...                     ...              ...             ...   
10122                 0.703            15476             117   
10123                 0.804             8764              69   
10124                 0.819            10291           

## Exploratory Data Analysis

In [22]:
#descriptive statistics
Bank.describe()

Unnamed: 0     CLIENTNUM  Customer_Age  Dependent_count  \
count  10127.000000  1.012700e+04  10127.000000     10127.000000   
mean    5063.000000  7.391776e+08     46.325960         2.346203   
std     2923.557422  3.690378e+07      8.016814         1.298908   
min        0.000000  7.080821e+08     26.000000         0.000000   
25%     2531.500000  7.130368e+08     41.000000         1.000000   
50%     5063.000000  7.179264e+08     46.000000         2.000000   
75%     7594.500000  7.731435e+08     52.000000         3.000000   
max    10126.000000  8.283431e+08     73.000000         5.000000   

       Months_on_book  Total_Relationship_Count  Months_Inactive_12_mon  \
count    10127.000000              10127.000000            10127.000000   
mean        35.928409                  3.812580                2.341167   
std          7.986416                  1.554408                1.010622   
min         13.000000                  1.000000                0.000000   
25%         31.000000                  3.000000                2.000000   
50%         36.000000                  4.000000                2.000000   
75%         40.000000                  5.000000                3.000000   
max         56.000000                  6.000000                6.000000   

       Contacts_Count_12_mon  Credit_Limit  Total_Revolving_Bal  \
count           10127.000000  10127.000000         10127.000000   
mean                2.455317   8631.938679          1162.814061   
std                 1.106225   9088.788539           814.987335   
min                 0.000000   1438.000000             0.000000   
25%                 2.000000   2555.000000           359.000000   
50%                 2.000000   4549.000000          1276.000000   
75%                 3.000000  11067.500000          1784.000000   
max                 6.000000  34516.000000          2517.000000   

       Avg_Open_To_Buy  Total_Amt_Chng_Q4_Q1  Total_Trans_Amt  Total_Trans_Ct  \
count     10127.000000          10127.000000     10127.000000    10127.000000   
mean       7469.139637              0.759941      4404.086304       64.858695   
std        9090.685324              0.219207      3397.129254       23.472570   
min           3.000000              0.000000       510.000000       10.000000   
25%        1324.500000              0.631000      2155.500000       45.000000   
50%        3474.000000              0.736000      3899.000000       67.000000   
75%        9859.000000              0.859000      4741.000000       81.000000   
max       34516.000000              3.397000     18484.000000      139.000000   

       Total_Ct_Chng_Q4_Q1  Avg_Utilization_Ratio  
count         10127.000000           10127.000000  
mean              0.712222               0.274894  
std               0.238086               0.275691  
min               0.000000               0.000000  
25%               0.582000               0.023000  
50%               0.702000               0.176000  
75%               0.818000               0.503000  
max               3.714000               0.999000

### From the above result, we get some important insights:

1. *Count:* All columns have a count of 10,127, indicating that there are no missing values in any of the columns.

2. *Mean (Average):*
   - The mean values provide a measure of central tendency for each column.
   - the mean of the second column is approximately 46.33, and the mean of the third column is approximately 2.35.

3. *Standard Deviation (Std):*
   - The standard deviation measures the amount of variation or dispersion in a set of values.
   - Larger standard deviations indicate greater variability.
   - For instance, the standard deviation of the second column is approximately 8.02.

4. *Minimum and Maximum:*
   - The minimum and maximum values show the range of each column.
   - the minimum age in the dataset is 26, and the maximum age is 73.

5. *Percentiles (25%, 50%, 75%):*
   - Percentiles provide information about the distribution of the data.
   - For instance, the 25th percentile (first quartile) of the second column (age) is 41, meaning that 25% of the individuals in the dataset are 41 years old or younger.